From 9c03bb14d2bd71f0fce4d54680028209e85fbe50 Mon Sep 17 00:00:00 2001
From: Daniel Gabay <daniel.gabay@intel.com>
Date: Tue, 7 Jan 2025 14:51:47 +0200
Subject: [PATCH] nl80211: Option to tell the driver about SPP A-MSDUs in assoc
 params

Use NL80211_ATTR_ASSOC_SPP_AMSDU attribute to indicate the driver
whether SPP A-MSDUs are used on this connection.

Signed-off-by: Daniel Gabay <daniel.gabay@intel.com>
---
 src/drivers/driver.h         | 6 +++++-
 src/drivers/driver_nl80211.c | 6 ++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/src/drivers/driver.h b/src/drivers/driver.h
index fb6b69ef4..f04050003 100644
--- a/src/drivers/driver.h
+++ b/src/drivers/driver.h
@@ -1397,11 +1397,15 @@ struct wpa_driver_associate_params {
 	 */
 	struct wpa_driver_mld_params mld_params;
 
-
 	/**
 	 * rsn_overriding - wpa_supplicant RSN overriding support
 	 */
 	bool rsn_overriding;
+
+	/**
+	 * spp_amsdu - SPP A-MSDU used on this connection
+	 */
+	bool spp_amsdu;
 };
 
 enum hide_ssid {
diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index d03c8f980..3aed4f3a9 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -7221,6 +7221,12 @@ static int nl80211_connect_common(struct wpa_driver_nl80211_data *drv,
 	    nla_put_flag(msg, NL80211_ATTR_MLO_SUPPORT))
 		return -1;
 
+	if (params->spp_amsdu) {
+		wpa_printf(MSG_DEBUG, "  * SPP A-MSDU");
+		if (nla_put_flag(msg, NL80211_ATTR_ASSOC_SPP_AMSDU))
+			return -1;
+	}
+
 	return 0;
 }
 
-- 
2.43.0

